lib/rules/no-extra-parens.js

```diff
@@ -75,6 +76,68 @@ module.exports = function(context) {
     }
 
     /**
+     * Checks whether or not a given node is located at the head of ExpressionStatement.
+     * @param {ASTNode} node - A node to check.
+     * @returns {boolean} `true` if the node is located at the head of ExpressionStatement.
+     */
+    function isHeadOfExpressionStatement(node) {
+        var parent = node.parent;
+        while (parent != null) {
+            switch (parent.type) {
+                case "SequenceExpression":
+                    if (parent.expressions[0] !== node || isParenthesised(node)) {
```

Comment:
Actually, I want to not do `isParenthesised(node)` if `parent.type` is `"ExpressionStatement"`. Because in `((function() {}).foo)` case, I want to not warn both parens.